static/app/components/searchQueryBuilder/valueCombobox.tsx

```diff
@@ -440,17 +441,17 @@ function cleanFilterValue(key: string, value: string): string {
       if (FILTER_VALUE_NUMERIC.test(value)) {
         return value;
       }
-      return '';
+      return null;
     case FieldValueType.INTEGER:
       if (FILTER_VALUE_INT.test(value)) {
         return value;
       }
-      return '';
+      return null;
     case FieldValueType.DATE:
       const parsed = parseFilterValueDate(value);
 
       if (!parsed) {
```

Comment:
super nit: parsed is only used once so we can put it in the conditional

Reply:
also  🙌  to swapping `''` for `null` to be more explicit about unmatched values.